.github/workflows/docker-build.yaml

```diff
@@ -0,0 +1,28 @@
+name: Docker Image Build
+
+on:
+    workflow_dispatch: {}
+    push:
+        paths-ignore:
+            - '.github/**'
+            - '.happy/**'
+            - 'dev_docs/**'
+            - 'example-dataset/**'
+        branches-ignore:
+          - main
```

Comment:
- only rebuild if the source changes
- don't build and push on the main branch (the container image should already be built by the time it merges to main)